e3sm_diags/driver/polar_driver.py

```diff
@@ -71,7 +71,7 @@ def run_diag(parameter: CoreParameter) -> CoreParameter:
             mask_path = os.path.join(
                 e3sm_diags.INSTALL_PATH, "acme_ne30_ocean_land_mask.nc"
             )
-            with cdms2.open(mask_path) as f:
+            with xcdat.open_dataset(mask_path) as f:
                 land_frac = f("LANDFRAC")
                 ocean_frac = f("OCNFRAC")
 
```

Comment:
This try, except statement should be replaced entirely by using `_get_land_sea_mask()`.

Example:
https://github.com/E3SM-Project/e3sm_diags/blob/59781c8ada30716d7d242b7d3a329daeeebbd81e/e3sm_diags/driver/lat_lon_driver.py#L61-L71

Reply:
@tomvothecoder If I make that change, then there are still 4 occurrences of `land_frac` and `ocean_frac` to deal with:

https://github.com/E3SM-Project/e3sm_diags/blob/8c7922d1581098291ddc4bec03480f984d71ec86/e3sm_diags/driver/polar_driver.py#L142-L147

https://github.com/E3SM-Project/e3sm_diags/blob/8c7922d1581098291ddc4bec03480f984d71ec86/e3sm_diags/driver/polar_driver.py#L207-L212

How do I address that issue? Can `land_frac` and `ocean_frac` be extracted from `xr.Dataset`? Or rather should the `MV` code be removed anyway in the refactor (as considered in https://github.com/E3SM-Project/e3sm_diags/pull/749#issuecomment-1870709997)?

Reply:
If you're replacing `land_frac` and `ocean_frac` (both `cdms2.TransientVariable`), then all subsequent references and functions used need to be replaced with their Xarray-based equivalent.

Please take a look at this [diff](https://github.com/E3SM-Project/e3sm_diags/compare/31dde4aa4b347e30b17a7057a2638bfcb8bce723#diff-6b1063c69e89b4f25f2ac5bbf631bbb77a904921f91784bcfd71ba24167df440) for the `lat_lon_driver.py` between the `cdat-migration-fy24` and `main` branches. `polar_drive.py` is pretty similar to `lat_lon_driver.py`, so you should be able to somewhat follow the diff.

In this case, `utils.general.utils.general.select_region()` and `utils.general.convert_to_pressure_levels()` have both been replaced by `utils.regrid.subset_align_datasets()`.

Reply:
For context, refactoring e3sm_diags is a not a simple 1:1 replacement of CDAT code with Xarray/xCDAT code. It requires updating the underlying data structures (cdms2 objects to Xarray objects) and replacing function calls to new functions that can operate Xarray objects. This is done all while making sure the newly refactored code is maintainable and easy to read (e.g,. reusing utility functions to eliminate code duplication), which is why the diff for `lat_lon_driver.py` is looks significant.